cpp/independent_set/CographIndependentSet.cpp

```diff
@@ -0,0 +1,68 @@
+#include <graph/GraphTools.hpp>
+
+#include "independent_set/CographIndependentSet.hpp"
+
+namespace Koala {
+    NetworKit::count CographIndependentSet::recurse_run(NetworKit::count n, NetworKit::count v) {
+        NetworKit::count i;
+        if (cotree->nodes[v].type == 2) {
+            return 1;
+        } else {
+            NetworKit::count l = 0, r = 0;
+            if (cotree->nodes[v].left_son != -1) {
+                l = recurse_run(n,cotree->nodes[v].left_son);
+            }
+
+            if (cotree->nodes[v].right_son != -1) {
+                r = recurse_run(n,cotree->nodes[v].right_son);
+            }
+
+            if (cotree->nodes[v].type == 1) {
+                return std::max(l, r);
+            } else {
+                return l + r;
+            }
+        }
+    }
+
+    void CographIndependentSet::run() {
+
+        if (cotree->prepared == false) {
+            cotree->BuildTree();
+        }
+        independet_set_size = recurse_run(n, n);
```

Comment:
Do you always pass n as the first parameter? If so, why can't you just get it via `G.numberOfNodes()`?